### ISSUE TITLE
After adding `gem 'capybara'` to gemfile, rspec cannot run

### ISSUE THREAD

**Issue by b-d-b** [jnicklas/capybara#1703]
I have the following test and development group in my Gemfile:
```

source 'https://rubygems.org'

gem 'rails', '4.2.5.1'
gem 'mysql2', '>= 0.3.13', '< 0.5'
gem 'sass-rails', '~> 5.0'
gem 'uglifier', '>= 1.3.0'
gem 'jquery-rails'
gem 'turbolinks'
gem 'jbuilder', '~> 2.0'
gem 'sdoc', '~> 0.4.0', group: :doc
gem 'devise'
gem 'cancancan', '~> 1.10'
gem 'unicorn' #server
gem 'whenever'
gem 'slim-rails'

group :development, :test do
  gem 'reek'
  gem 'byebug'
  gem 'pry-rails'
  gem 'pry-byebug'
  gem 'spring-commands-rspec'
  gem 'guard-rspec'
  gem 'rspec-rails'
  gem 'capybara'
  gem 'factory_girl_rails'
  gem 'database_cleaner'
  gem 'shoulda-matchers', require: false
  gem 'simplecov', group: :test
end

group :development do
  gem 'rubocop', require: false
  gem 'web-console', '~> 2.0'
  gem 'spring'
  gem 'capistrano'
  gem 'capistrano-rails'
  gem 'capistrano-bundler'
  gem 'capistrano-rvm'
  gem 'capistrano3-unicorn'
end
```

After install, and before  I made any changes to my spec_helper or rails_helper files, I could not run guard-rspec anymore.  I got the following error:

```
.rvm/rubies/ruby-2.2.1/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/builder.rb:146:in `to_app': missing run or map statement (RuntimeError)
```

Any leads on why this might happen?

**Comment by twalpole** [comment]
No clue -- look further up the stacktrace to see where the error is coming from.  Please ask howto question in the mailing list as requested at the top of the README

**Comment by lassiter** [comment]
I also got this error.

This is the trace when trying to run `rspec spec`
```
/Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rack-2.0.4/lib/rack/builder.rb:146:in `to_app': missing run or map statement (RuntimeError)
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rack-2.0.4/lib/rack/builder.rb:160:in `block in generate_map'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rack-2.0.4/lib/rack/builder.rb:160:in `each'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rack-2.0.4/lib/rack/builder.rb:160:in `generate_map'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rack-2.0.4/lib/rack/builder.rb:145:in `to_app'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/capybara-3.0.0/lib/capybara/rails.rb:9:in `<top (required)>'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:292:in `require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:292:in `block in require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:258:in `load_dependency'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:292:in `require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-rails-3.7.2/lib/rspec/rails/vendor/capybara.rb:7:in `<top (required)>'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:292:in `require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:292:in `block in require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:258:in `load_dependency'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.5/lib/active_support/dependencies.rb:292:in `require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-rails-3.7.2/lib/rspec/rails.rb:15:in `<top (required)>'
        from /Users/lassitergregg/Desktop/bloc/ruby/blocipedia/spec/spec_helper.rb:4:in `require'
        from /Users/lassitergregg/Desktop/bloc/ruby/blocipedia/spec/spec_helper.rb:4:in `<top (required)>'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration.rb:1455:in `require'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration.rb:1455:in `block in requires='
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration.rb:1455:in `each'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration.rb:1455:in `requires='
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration_options.rb:112:in `block in process_options_into'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration_options.rb:111:in `each'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration_options.rb:111:in `process_options_into'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/configuration_options.rb:21:in `configure'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:99:in `setup'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:86:in `run'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:71:in `run'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:45:in `invoke'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/gems/rspec-core-3.7.1/exe/rspec:4:in `<top (required)>'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/bin/rspec:23:in `load'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/bin/rspec:23:in `<main>'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `eval'
        from /Users/lassitergregg/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `<main>'
```

This is my gem file
```
source 'https://rubygems.org'

git_source(:github) do |repo_name|
  repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
  "https://github.com/#{repo_name}.git"
end


# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '~> 5.1.5'
# Use Puma as the app server
gem 'puma', '~> 3.7'
# Use SCSS for stylesheets
gem 'sass-rails', '~> 5.0'
# Bootstrap sass
gem 'bootstrap'
gem 'sprockets-rails', '>= 3.2.1'
# Bootstrap Dependency
gem 'jquery-rails'
# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'
# See https://github.com/rails/execjs#readme for more supported runtimes
# gem 'therubyracer', platforms: :ruby

# Use CoffeeScript for .coffee assets and views
gem 'coffee-rails', '~> 4.2'
# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
gem 'turbolinks', '~> 5'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.5'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'

# Devise is a flexible authentication solution for Rails based on Warden.
gem 'devise'
# Pundit is a set of helpers which guide you in leveraging regular Ruby classes and object oriented design patterns to build a simple, robust and scaleable authorization system.
gem "pundit"

# Use ActiveModel has_secure_password
gem 'bcrypt', '~> 3.1.7'

# Stripe - Payment Gateway
gem 'stripe'

group :production do
  # Use pg as the production database for Active Record
  gem 'pg', '~> 0.20'
  gem 'rails_12factor'
end

group :development, :test do
  gem 'rspec-rails', '~> 3.0'
  gem 'rails-controller-testing'
  gem 'shoulda'
  gem 'factory_bot_rails'
  gem 'faker'
end
group :test do
  gem "capybara"
end
group :development do
  gem 'sqlite3'
  gem 'pry-rails'
  gem 'ruby-debug-ide'
  gem 'debase'
  # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
  gem 'web-console', '>= 3.3.0'
  gem 'listen', '>= 3.0.5', '< 3.2'
  # using mailcatcher for emails
end

# Used for ENV variables
gem 'figaro', '1.0'

```

**Comment by twalpole** [comment]
@lassiter From the stack trace it looks like you're requiring `rspec/rails` (rspec-rails gem) from your `spec_helper.rb` instead of from `rails_helper.rb` where it's supposed to be loaded from (see installation instructions - https://github.com/rspec/rspec-rails#installation ).  I'm guessing this means it's being loaded before the Rails application has been initialized.  In the future it's probably best not to comment on a closed issue from 2 years ago.

**Comment by bensheldon** [comment]
For anyone else who arrives at this error through searching (as I did) for the exception raised by ```rack/builder.rb:146:in `to_app': missing run or map statement (RuntimeError)```. It is triggered by this code when `Rails.application` is `nil`:

https://github.com/teamcapybara/capybara/blob/c0cbf4024c1abd48b0c22c2930e7b05af58ab284/lib/capybara/rails.rb#L5-L9

`Rails.application` can be `nil` when this file in Capybara is loaded and one has not loaded/declared their `class Application < Rails::Application` (typically by loading `config/application.rb`). 

This situation can only arise out of a misconfiguration. I came to it via Tapioca, which will somewhat randomly load files without order, leading to this error.
